text/src/swam/witx/parser/ImportContext.scala

```diff
@@ -0,0 +1,54 @@
+package swam.witx.parser
+
+import java.io.File
+import java.nio.file.{Files, NoSuchFileException, Path, Paths}
+
+import cats.effect._
+import fastparse.Parsed
+import swam.text.{ParserException, readFile}
+import swam.witx.WitxParser
+import swam.witx.unresolved.{AliasType, BaseWitxType, ModuleInterface}
+
+import scala.concurrent.ExecutionContext
+
+/**
+  *@author Javier Cabrera-Arteaga on 2020-03-18
+  */
+class ImportContext[F[_]](val includes: List[String])(implicit val F: Effect[F]) {
+
+  // TODO FIX !
+  implicit val cs = IO.contextShift(ExecutionContext.Implicits.global)
+
+  def load(path: String) = {
+
+    val paths = getInPath(path)
+
+    if (paths.isEmpty)
+      throw new NoSuchFileException(path)
+
+    Blocker[IO]
+      .use { blocker =>
+        WitxParser[IO].parseTypes(paths.head, blocker)
+      }
+      .unsafeRunSync()
+  }
+
+  def getInPath(path: String): List[Path] = {
+    if (Files.exists(Paths.get(path)))
+      List(Paths.get(path)) // Current working directory
+    else // else search in path
+      System
+        .getenv("PATH")
+        .split(":")
```

Comment:
This will raise a NPE if `PATH` is not defined. Prefer using the scala [`sys.env`](https://www.scala-lang.org/api/current/scala/sys/index.html#env:Map[String,String]) map with the appropriate `get` method, which returns an `Option`.